.github/workflows/reusable-ubuntu.yml

```diff
@@ -34,9 +34,12 @@ jobs:
       PYTHONSTRICTEXTENSIONBUILD: 1
       TERM: linux
     steps:
-    - uses: actions/checkout@v4
+    - uses: Wandalen/wretry.action@master
```

Comment:
Is there a feature request for `actions/checkout` to retry checkouts?

I'm unsure about using a third party action here, vs e.g. a `run:` block just using `git` commands.

Reply:
We raised a support ticket with GitHub and a couple of tips have been generated. One of this was to add a `retry` to the checkout.

![image](https://github.com/user-attachments/assets/8fb0eaca-9e16-4dcc-8cfb-59d3916e1288)

Funny enough, the retry doesn't exist on the checkout action!

I'm going to raise one

Reply:
Raised: https://github.com/actions/checkout/issues/2048

Reply:
Re: https://github.com/diegorusso/cpython/actions/runs/12948283932/job/36116629838#step:4:157

That's just the third-party linter that complaining about something that either doesn't exist or hasn't been added to the linter's schema.

Please could you rename your branch to `3.14-test-retry` so it will trigger some jobs that use actions/checkout?

Reply:
Same thing: https://github.com/diegorusso/cpython/actions/runs/12949425398/job/36120135892

BTW also the GitHub editor was complaining about the retry.

<img width="319" alt="Screenshot 2025-01-24 at 12 17 09" src="https://github.com/user-attachments/assets/f66ece7c-fcf9-4dd1-b449-8faaa53ea9a4" />

Reply:
I think here we are looking at this from the wrong angle. If you see the failures posted initially, they don't fail while checking out the repo.
It is not started at all. After it prints the env variable there is nothing. I have the feeling that there is some infrastructure issue in running that action on a specific node.
Even if we put the retry 100 times, they won't be executed as this fails even before to start the process.

Reply:
Yes, that's the impression I get too.

Looking more closely at the raw logs (https://github.com/python/cpython/actions/runs/12934438169/job/36075486419?pr=129232 > ⚙️ > View raw logs), the checkout does actually succeed. 

For some reason, the next step just doesn't start.

Comparing with a [passing build](https://github.com/python/cpython/actions/runs/12950469935/job/36123368902?pr=128761):

<img width="2553" alt="image" src="https://github.com/user-attachments/assets/d058930c-7541-4638-80e3-bb718b039cdc" />

Reply:
Oh wow, I didn't remember that we had access to these raw logs. They are useful indeed.
So the retry is pointless because it actually it doesn't fail on the checkout.